tests/drop_counters/test_drop_counters.py

```diff
@@ -185,11 +331,29 @@ def log_pkt_params(dut_iface, mac_dst, mac_src, ip_dst, ip_src):
     logger.info("Packet IP SRC - {}".format(ip_src))
 
 
-def base_verification(discard_group, pkt, ptfadapter, duthost, combined_counter, ptf_tx_port_id, dut_iface):
-    """
-    Base test function for verification of L2 or L3 packet drops. Verification type depends on 'discard_group' value.
-    Supported 'discard_group' values: 'L2', 'L3'
-    """
+def ensure_no_l3_drops(duthost):
+    """ Verify L3 drop counters were not incremented """
+    intf_l3_counters = get_pkt_drops(duthost, GET_L3_COUNTERS)
+    unexpected_drops = {}
+    for iface, value in intf_l3_counters.items():
+        if int(value[RX_ERR]) >= PKT_NUMBER:
+            unexpected_drops[iface] = int(value[RX_ERR])
+    if unexpected_drops:
+        pytest.fail("L3 'RX_ERR' was incremented for the following interfaces:\n{}".format(unexpected_drops))
+
+
+def ensure_no_l2_drops(duthost):
+    """ Verify L2 drop counters were not incremented """
+    intf_l2_counters = get_pkt_drops(duthost, GET_L2_COUNTERS)
+    unexpected_drops = {}
+    for iface, value in intf_l2_counters.items():
+        if int(value[RX_DRP]) >= PKT_NUMBER:
+            unexpected_drops[iface] = int(value[RX_DRP])
+    if unexpected_drops:
+        pytest.fail("L2 'RX_DRP' was incremented for the following interfaces:\n{}".format(unexpected_drops))
```

Comment:
If we combine this in one function, the code inside function will be mostly the same except we can add if l2 -> code from "ensure_no_l2_drops"
if l3 -> code from "ensure_no_l3_drops".
Because there is used differnt function to get counters, different keys, and different logging message for L2 or L3.

In such case the benefit will be only where this function will be called:
1. ensure_no_l3_drops or ensure_no_l2_drops
vs
2. ensure_no_drops("L2") or ensure_no_drops("L3")

In such case I think this can be as it is.

Thoughts?

Reply:
I totally missed the RX_ERR vs. RX_DRP the first run through. I think you're right, this is OK as is. :)